### ISSUE TITLE
CacheCow.Server45 doesn't produce ETag for Redis storage.

### ISSUE THREAD

**Issue by studenikin** [aliostad/CacheCow#187]
We have the following packages installed:
* CacheCow.Server45
* CacheCow.Server.EntityTagStore.Redis45
* CacheCow.Common45

The configuration:
```
    public static class WebApiConfig
    {
        public static void Register(HttpConfiguration config)
        {
            config.MapHttpAttributeRoutes();

            var redisTagStore = new RedisEntityTagStore("redis1.com:6379,redis2.com:6379,password=12345", 0, TimeSpan.FromSeconds(10));

            config.MessageHandlers.Add(new CachingHandler(config, redisTagStore));
        }
    }
}
```
Etag is not generated when requesting controller.
Although when I switch back to 4.0 version of CacheCow everything starts working OK.

**Comment by aliostad** [comment]
It is definitely getting generated - just created a test project. It is possible you have something else in your pipeline that removes it?

**Comment by studenikin** [comment]
I just created empty WebApi solution, and the tag doesn't work.

Have you tested against Redis store?
When I'm testing against memory store it is working fine. But Redis store gives the issue.

**Comment by aliostad** [comment]
No I did not use Redis store. I will just do that.

**Comment by aliostad** [comment]
OK just ran and it returns fine.
![image](https://user-images.githubusercontent.com/487334/28137512-74372b8e-6745-11e7-8b7b-c3c1b8b4f2f7.png)

**Comment by studenikin** [comment]
That's weird, since ETag sometimes appears, but most of the time it doesn't.
I debugged the sources and it seems the problem might occur in this place:

```
internal Func<HttpResponseMessage, HttpResponseMessage> GetCachingContinuation(HttpRequestMessage request)
{
	return response =>
	{
		if (!response.IsSuccessStatusCode) // only if successful carry on processing
			return response;

		try
		{
			var cacheKey = GenerateCacheKey(request);
			ExecuteCacheInvalidationRules(cacheKey, request, response);
			ExecuteCacheAdditionRules(cacheKey, request, response);
			return response;
		}
		catch (Exception ex)
		{
		   Trace.TraceWarning("Exception in CacheCow: " + ex.ToString());
			return response;
		}

	   
	};
}
```

My guess is that because the ExecuteCacheAdditionRules is not awaited, the response may be returned before the ExecuteCacheAdditionRules fills it out.
This is happening in my case. I'm getting no Etag in 9 cases out of 10.

**Comment by aliostad** [comment]
Wow... You found a nasty bug. Thank you!!

**Comment by aliostad** [comment]
When I moved everything to async, one function stayed. Great spot. I checked the rest, it seems all fine.

**Comment by aliostad** [comment]
Fixed with v1.1.0 of 45 packages. Can you please test them? This should be fixed now. Closing the issue but can re-open if needs be,

**Comment by studenikin** [comment]
Tested. Works fine. Thanks!
